### PR TITLE
BookA-Pset_LabTestResult

### PR DESCRIPTION
**Pull request by larswik**
# Pset: Pset_LabTestResult

## Description: 

Properties resulting from commonly conducted laboratory tests on soil, rock or fluid specimens.

| Property | Datatype | Description |
|----------|----------|-------------|
|TestedMaterial|IfcLabel|Description of the tested material, e.g. according to existing rock and soil classification systems.|
|UCS|IfcPressureMeasure|The unconfined compressive strength (UCS) is the maximum axial compressive stress that a right-cylindrical sample of material can withstand under unconfined conditions.|
|YoungModulus|According to existing definition|According to existing definition|
|Density|IfcMassDensityMeasure|The weight per unit volume of a substance.|
|WaterContent|IfcPositiveRatioMeasure|The ratio of weight of water Ww to the weight of solids (Ws or Wd) in a given mass of soil or rock.|
|Porosity|IfcPositiveRatioMeasure|The volumetric fraction of pores in the material.|
|CohesionBehaviour| According to existing definition. | According to existing definition. |
|FrictionAngle|IfcPlaneAngleMeasure|Derived from the Mohr-Coulomb failure criterion and used to describe the friction shear resistance of ground materials, together with the normal effective stress.|
|LiquidLimits|IfcPositiveRatioMeasure|Liquid limit is the water content at which a pat of soil, cut by a groove of standard dimension, will flow together for a distance of 13 mm under the impact of 25 drops by a standard liquid limit device.Also, can definition as the water content corresponding to any arbitrary limit between the liquid and plastic consistency states of a soil. |
|PlasticLimit|IfcPositiveRatioMeasure|Water content at which the soil changes from the plastic state to a semi-solid state.|
|PlasticityIndex|IfcPositiveRatioMeasure|The plasticity index Ip is the numerical difference between the liquid limit and plastic limit of a fine soil.  IP = wL - wP.|
|Slaking|IfcLabel|The crumbling and disintegration of rock or hard soil upon exposure to air or water.|
|HydraulicConductivity|IfcVelocityMeasure|Hydraulic Conductivity (permeability) of soil for water, given with the K or Kf value in m/s.|

### CODE REVIEW

**Comment by patmclarin** [comment]
can we update the Pset description to: Properties resulting from commonly conducted laboratory tests on soil, rock or fluid specimens

File: IFC4x3/Properties/d/Density_2DPZa5h0XAWe3U36TQs6B7/DocProperty.xml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="Density_2DPZa5h0XAWe3U36TQs6B7" Name="Density" UniqueId="8d663905-ac08-4a82-80de-0c675ad862c7" PrimaryDataType="IfcMassDensityMeasure" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/d/Density_2DPZa5h0XAWe3U36TQs6B7/DocProperty.xml]: Use existing `MassDensity`?

File: IFC4x3/Properties/e/Emodulus_3hpNJVftrE1AuFMdtJ4v6C/DocProperty.xml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="Emodulus_3hpNJVftrE1AuFMdtJ4v6C" Name="Emodulus" UniqueId="ebcd74df-a77d-4e04-ae0f-5a7dd313918c" PrimaryDataType="IfcPressureMeasure" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/e/Emodulus_3hpNJVftrE1AuFMdtJ4v6C/DocProperty.xml]: Use existing `YoungModulus`.

File: IFC4x3/Properties/h/HydraulicConductivity_19kpZnztn1VQFucvmNUC96/DocProperty.xml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="HydraulicConductivity_19kpZnztn1VQFucvmNUC96" Name="HydraulicConductivity" UniqueId="49bb38f1-f77c-417d-a3f8-9b9c1778c246" PrimaryDataType="IfcVelocityMeasure" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/h/HydraulicConductivity_19kpZnztn1VQFucvmNUC96/DocProperty.xml]: Not too happy about `IfcVelocityMeasure` here. How about `IfcVaporPermeabilityMeasure`? Not the same but at least it is about water/vapor permeability.

File: IFC4x3/Properties/l/LiquidLimits_2AHK69qTTB1htxysz9ebsl/Documentation.md [hunk 4]
```diff
@@ -0,0 +1 @@
+Liquid limit is the water content at which a pat of soil, cut by a groove of standard dimension, will flow together for a distance of 13 mm under the impact of 25 drops by a standard liquid limit device.Also, can definition as the water content corresponding to any arbitrary limit between the liquid and plastic consistency states of a soil.
```
Comment (SergejMuhic) [inline IFC4x3/Properties/l/LiquidLimits_2AHK69qTTB1htxysz9ebsl/Documentation.md]: Last sentence is weird. Please correct.

```suggestion
Liquid limit is the water content at which a pat of soil, cut by a groove of standard dimension, will flow together for a distance of 13 mm under the impact of 25 drops by a standard liquid limit device. Also, can definition as the water content corresponding to any arbitrary limit between the liquid and plastic consistency states of a soil.
```

File: IFC4x3/Properties/p/PlasticLimit_2QRKg6a61Ft9rL$4_1Q7GD/DocProperty.xml [hunk 5]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="PlasticLimit_2QRKg6a61Ft9rL4_1Q7GD" Name="PlasticLimit" UniqueId="9a6d4a86-9060-4fdc-9d55-fc4f8168740d" PrimaryDataType="IfcPositiveRatioMeasure" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/p/PlasticLimit_2QRKg6a61Ft9rL$4_1Q7GD/DocProperty.xml]: Ratio measure?

File: IFC4x3/Properties/p/PlasticityIndex_0$zff_imHDMetkQE35AOmz/Documentation.md [hunk 6]
```diff
@@ -0,0 +1 @@
+The plasticity index Ip is the numerical difference between the liquid limit and plastic limit of a fine soil.  IP = wL ? wP.
```
Comment (SergejMuhic) [inline IFC4x3/Properties/p/PlasticityIndex_0$zff_imHDMetkQE35AOmz/Documentation.md]: What does this "IP = wL ? wP." mean? Should it be "IP = wL / wP."?

File: IFC4x3/Properties/s/Slaking_1GTIyqmf9Eu8cGgYc0$TFp/DocProperty.xml [hunk 7]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="Slaking_1GTIyqmf9Eu8cGgYc0TFp" Name="Slaking" UniqueId="50752f34-c292-4ee0-8990-aa2980fdd3f3" PrimaryDataType="IfcLabel" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/s/Slaking_1GTIyqmf9Eu8cGgYc0$TFp/DocProperty.xml]: `IfcLabel`?

File: IFC4x3/Properties/u/UCS_2ZCp5dntj43BvWF_dlBh7I/DocProperty.xml [hunk 8]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
+<DocProperty xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" id="UCS_2ZCp5dntj43BvWF_dlBh7I" Name="UCS" UniqueId="a3333167-c77b-440c-be60-3fe9ef2eb1d2" PrimaryDataType="IfcPressureMeasure" />
```
Comment (SergejMuhic) [inline IFC4x3/Properties/u/UCS_2ZCp5dntj43BvWF_dlBh7I/DocProperty.xml]: Propose renaming to `UnconfinedCompressiveStrength`.

File: IFC4x3/Properties/d/Density_2DPZa5h0XAWe3U36TQs6B7/DocProperty.xml (hunk 1 above)
Comment (larswik) [reply IFC4x3/Properties/d/Density_2DPZa5h0XAWe3U36TQs6B7/DocProperty.xml]: > Use existing `MassDensity`?

That should be fine! Done in [53b1364](https://github.com/bSI-InfraRoom/IFC-Specification/pull/612/commits/53b136489c1be39e04ee4b4b8f0e57258e6f6be0)

File: IFC4x3/Properties/e/Emodulus_3hpNJVftrE1AuFMdtJ4v6C/DocProperty.xml (hunk 2 above)
Comment (larswik) [reply IFC4x3/Properties/e/Emodulus_3hpNJVftrE1AuFMdtJ4v6C/DocProperty.xml]: > Use existing `YoungModulus`.

If it is tensile strength. Need to ask @j-weil if this is ok.

File: IFC4x3/Properties/h/HydraulicConductivity_19kpZnztn1VQFucvmNUC96/DocProperty.xml (hunk 3 above)
Comment (larswik) [reply IFC4x3/Properties/h/HydraulicConductivity_19kpZnztn1VQFucvmNUC96/DocProperty.xml]: > Not too happy about `IfcVelocityMeasure` here. How about `IfcVaporPermeabilityMeasure`? Not the same but at least it is about water/vapor permeability.

The definition mentions "given with the K or Kf value in **m/s**". Vapor permeability seems to be measured in kg / (s * m2 * Pa). @j-weil

File: IFC4x3/Properties/l/LiquidLimits_2AHK69qTTB1htxysz9ebsl/Documentation.md (hunk 4 above)
Comment (larswik) [reply IFC4x3/Properties/l/LiquidLimits_2AHK69qTTB1htxysz9ebsl/Documentation.md]: @j-weil.

File: IFC4x3/Properties/p/PlasticLimit_2QRKg6a61Ft9rL$4_1Q7GD/DocProperty.xml (hunk 5 above)
Comment (larswik) [reply IFC4x3/Properties/p/PlasticLimit_2QRKg6a61Ft9rL$4_1Q7GD/DocProperty.xml]: > Ratio measure?

It is currently defined as IfcPositiveRatioMeasure. Is 0 is a valid value?

File: IFC4x3/Properties/p/PlasticityIndex_0$zff_imHDMetkQE35AOmz/Documentation.md (hunk 6 above)
Comment (larswik) [reply IFC4x3/Properties/p/PlasticityIndex_0$zff_imHDMetkQE35AOmz/Documentation.md]: > What does this "IP = wL ? wP." mean? Should it be "IP = wL / wP."?

This must have happened in the excel=>IfcDoc transformations. It should be wL - wP. Fixed in [b5effb1](https://github.com/bSI-InfraRoom/IFC-Specification/pull/612/commits/b5effb1445ea2d282b2324e89f204a04328fe038)

File: IFC4x3/Properties/s/Slaking_1GTIyqmf9Eu8cGgYc0$TFp/DocProperty.xml (hunk 7 above)
Comment (larswik) [reply IFC4x3/Properties/s/Slaking_1GTIyqmf9Eu8cGgYc0$TFp/DocProperty.xml]: > `IfcLabel`?

According to excel spec.

File: IFC4x3/Properties/u/UCS_2ZCp5dntj43BvWF_dlBh7I/DocProperty.xml (hunk 8 above)
Comment (larswik) [reply IFC4x3/Properties/u/UCS_2ZCp5dntj43BvWF_dlBh7I/DocProperty.xml]: > Propose renaming to `UnconfinedCompressiveStrength`.

UCS is compact and maybe well known by the domain? @j-weil.

File: IFC4x3/Properties/e/Emodulus_3hpNJVftrE1AuFMdtJ4v6C/DocProperty.xml (hunk 2 above)
Comment (larswik) [reply IFC4x3/Properties/e/Emodulus_3hpNJVftrE1AuFMdtJ4v6C/DocProperty.xml]: Fixed in [25580c9](https://github.com/bSI-InfraRoom/IFC-Specification/pull/612/commits/25580c952e660a7aafce1122cb397281016824ed)

File: IFC4x3/Properties/e/Emodulus_3hpNJVftrE1AuFMdtJ4v6C/DocProperty.xml [hunk 9]
```diff
@@ -0,0 +1,3 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (SergejMuhic) [inline IFC4x3/Properties/e/Emodulus_3hpNJVftrE1AuFMdtJ4v6C/DocProperty.xml]: Will remove

File: IFC4x3/Properties/e/Emodulus_3hpNJVftrE1AuFMdtJ4v6C/DocProperty.xml (hunk 2 above)
Comment (SergejMuhic) [reply IFC4x3/Properties/e/Emodulus_3hpNJVftrE1AuFMdtJ4v6C/DocProperty.xml]: This is resolved, right?

File: IFC4x3/Properties/h/HydraulicConductivity_19kpZnztn1VQFucvmNUC96/DocProperty.xml (hunk 3 above)
Comment (SergejMuhic) [reply IFC4x3/Properties/h/HydraulicConductivity_19kpZnztn1VQFucvmNUC96/DocProperty.xml]: Yes, even if units are the same, the measure is not. It is measured differently than linear velocity.

File: IFC4x3/Properties/s/Slaking_1GTIyqmf9Eu8cGgYc0$TFp/DocProperty.xml (hunk 7 above)
Comment (SergejMuhic) [reply IFC4x3/Properties/s/Slaking_1GTIyqmf9Eu8cGgYc0$TFp/DocProperty.xml]: Okay, in that case this is resolved, right?

File: IFC4x3/Properties/u/UCS_2ZCp5dntj43BvWF_dlBh7I/DocProperty.xml (hunk 8 above)
Comment (SergejMuhic) [reply IFC4x3/Properties/u/UCS_2ZCp5dntj43BvWF_dlBh7I/DocProperty.xml]: We are defining properties for the IFC domain though. 😄

File: IFC4x3/Properties/e/Emodulus_3hpNJVftrE1AuFMdtJ4v6C/DocProperty.xml (hunk 9 above)
Comment (SergejMuhic) [reply IFC4x3/Properties/e/Emodulus_3hpNJVftrE1AuFMdtJ4v6C/DocProperty.xml]: Removed.

File: IFC4x3/Properties/h/HydraulicConductivity_19kpZnztn1VQFucvmNUC96/DocProperty.xml (hunk 3 above)
Comment (larswik) [reply IFC4x3/Properties/h/HydraulicConductivity_19kpZnztn1VQFucvmNUC96/DocProperty.xml]: Keep velocity according to domain experts.

File: IFC4x3/Properties/l/LiquidLimits_2AHK69qTTB1htxysz9ebsl/Documentation.md (hunk 4 above)
Comment (larswik) [reply IFC4x3/Properties/l/LiquidLimits_2AHK69qTTB1htxysz9ebsl/Documentation.md]: Changed according to domain proposal.

File: IFC4x3/Properties/p/PlasticLimit_2QRKg6a61Ft9rL$4_1Q7GD/DocProperty.xml (hunk 5 above)
Comment (larswik) [reply IFC4x3/Properties/p/PlasticLimit_2QRKg6a61Ft9rL$4_1Q7GD/DocProperty.xml]: Description updated according to proposal from domain.

File: IFC4x3/Properties/s/Slaking_1GTIyqmf9Eu8cGgYc0$TFp/DocProperty.xml (hunk 7 above)
Comment (larswik) [reply IFC4x3/Properties/s/Slaking_1GTIyqmf9Eu8cGgYc0$TFp/DocProperty.xml]: IfcLabel it is.

File: IFC4x3/Properties/u/UCS_2ZCp5dntj43BvWF_dlBh7I/DocProperty.xml (hunk 8 above)
Comment (larswik) [reply IFC4x3/Properties/u/UCS_2ZCp5dntj43BvWF_dlBh7I/DocProperty.xml]: UnconfiedCompressiveStrength it is.

File: IFC4x3/Properties/p/PlasticLimit_2QRKg6a61Ft9rL$4_1Q7GD/DocProperty.xml (hunk 5 above)
Comment (SergejMuhic) [reply IFC4x3/Properties/p/PlasticLimit_2QRKg6a61Ft9rL$4_1Q7GD/DocProperty.xml]: > > Ratio measure?
> 
> It is currently defined as IfcPositiveRatioMeasure. Is 0 is a valid value?

0 is not valid.

**Pull request merged**